# Task 5: Statistical Analysis of Data

## Objective
Perform statistical analysis on the Student Academic Performance & Attendance Dataset using Python, NumPy, and SciPy.

## Tools Used
- Python
- Pandas
- NumPy
- SciPy

## Dataset
Student Academic Performance & Attendance Dataset

In [1]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import ttest_ind

# Importing Dataset

In this step, we load the dataset and inspect the first few records.

In [2]:
df = pd.read_csv("student_academic_performance.csv")

df.head()

,Math_Score,English_Score,Science_Score,Attendance
0,79.967142,86.792265,72.598574,75.460962
1,73.617357,81.095604,76.843851,80.698075
2,81.476885,70.715564,71.660641,82.931972
3,90.230299,62.236759,75.536308,94.438438
4,72.658466,78.378680,62.851083,87.782766


# Dataset Information

Checking the structure, data types, and dimensions of the dataset.

In [3]:
print("Dataset Shape:", df.shape)

df.info()

Dataset Shape: (1005, 4)
<class 'pandas.DataFrame'>
RangeIndex: 1005 entries, 0 to 1004
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Math_Score     1005 non-null   float64
 1   English_Score  1005 non-null   float64
 2   Science_Score  1005 non-null   float64
 3   Attendance     1005 non-null   float64
dtypes: float64(4)
memory usage: 31.5 KB


# Missing Value Analysis

Checking whether the dataset contains any missing values.

In [4]:
df.isnull().sum()

Math_Score       0
English_Score    0
Science_Score    0
Attendance       0
dtype: int64

# Descriptive Statistics

Generating summary statistics for numerical columns.

In [5]:
df.describe()

,Math_Score,English_Score,Science_Score,Attendance
count,1005.000000,1005.000000,1005.000000,1005.000000
mean,75.199347,70.873287,78.024753,84.887303
std,9.825647,12.061808,7.901797,5.179065
min,45.060838,34.331166,53.609156,68.820336
25%,68.534271,62.731421,72.801878,81.237824
50%,75.264884,70.756926,77.951531,84.976378
75%,81.476885,78.818533,83.273965,88.327084
max,105.378939,107.424099,102.433148,100.897548


# Interpretation of Descriptive Statistics

The dataset contains 1005 student records.

## Math Score
- Mean: 75.20
- Median: 75.26
- Standard Deviation: 9.83
- Minimum Score: 45.06
- Maximum Score: 105.38

## English Score
- Mean: 70.87
- Median: 70.76
- Standard Deviation: 12.06
- Minimum Score: 34.33
- Maximum Score: 107.42

## Science Score
- Mean: 78.02
- Median: 77.95
- Standard Deviation: 7.90
- Minimum Score: 53.61
- Maximum Score: 102.43

## Attendance
- Mean Attendance: 84.89%
- Median Attendance: 84.98%
- Standard Deviation: 5.18
- Minimum Attendance: 68.82%
- Maximum Attendance: 100.90%

## Observation
The average Science Score is the highest among all subjects, while English Score has the highest variation as indicated by its standard deviation. Attendance levels are generally high, with an average attendance of approximately 85%.

# Mean, Median and Mode Analysis

Mean represents the average value of the data.

Median represents the middle value when data is arranged in ascending order.

Mode represents the most frequently occurring value in the dataset.

In [7]:
scores = df["Math_Score"]

print("Mean =", scores.mean())
print("Median =", scores.median())
print("Mode =", scores.mode()[0])

Mean = 75.19934650846717
Median = 75.26488444602683
Mode = 105.37893945223034


# Mean, Median and Mode Analysis

## Results for Math Score

- Mean = 75.20
- Median = 75.26
- Mode = 105.38

## Interpretation

The mean Math Score is 75.20, indicating the average performance of students in Mathematics.

The median score is 75.26, which is very close to the mean. This suggests that the data is approximately symmetric and does not contain significant skewness.

The mode score is 105.38, indicating the most frequently occurring Math Score in the dataset.

## Conclusion

Since the mean and median values are very close, the distribution of Math Scores appears to be fairly balanced. Students achieved an average score of approximately 75 marks in Mathematics.

# T-Test Analysis

A T-Test is performed to determine whether there is a statistically significant difference in Math Scores between students with high attendance and low attendance.

## Hypotheses

### Null Hypothesis (H₀)
There is no significant difference in Math Scores between students with high attendance and low attendance.

### Alternative Hypothesis (H₁)
There is a significant difference in Math Scores between students with high attendance and low attendance.

In [8]:
from scipy.stats import ttest_ind

high_attendance = df[df['Attendance'] >= 85]['Math_Score']
low_attendance = df[df['Attendance'] < 85]['Math_Score']

t_statistic, p_value = ttest_ind(high_attendance, low_attendance)

print("T Statistic =", t_statistic)
print("P Value =", p_value)

T Statistic = -0.426491875652055
P Value = 0.6698409850384432


# T-Test Results

## Results

- T Statistic = -0.4265
- P Value = 0.6698

## Interpretation

The p-value obtained from the T-Test is 0.6698, which is greater than the significance level of 0.05.

Therefore, we fail to reject the Null Hypothesis.

# Conclusion

Based on the statistical analysis:

- Mean, Median, and Mode were calculated for the Math Score.
- A T-Test was performed to compare Math Scores between students with high attendance and low attendance.
- The obtained p-value (0.6698) is greater than 0.05.

Therefore, there is no statistically significant difference in Math Scores between the two attendance groups.

## Final Outcome

The statistical analysis was successfully performed using Python, NumPy, Pandas, and SciPy. The dataset was analyzed quantitatively using descriptive statistics and hypothesis testing techniques.